sbt-web-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/WebScalaJSBundlerPlugin.scala

```diff
@@ -60,19 +60,25 @@ object WebScalaJSBundlerPlugin extends AutoPlugin {
     )
   }
 
-  def bundlesWithSourceMaps(sjsStage: TaskKey[Attributed[File]]): Def.Initialize[Task[Seq[(File, String)]]] =
+  val bundlesWithSourceMaps: Def.Initialize[Task[Seq[(File, String)]]] =
     Def.taskDyn {
       val projects = scalaJSProjects.value.map(Project.projectToRef)
       Def.task {
         // ((file, relative-path), sourceMapsEnabled)
         val bundles: Seq[((File, String), Boolean)] =
           projects
             .map { project =>
-              Def.task {
-                val files = (webpack in (project, Compile, sjsStage in project)).value
-                val clientTarget = (npmUpdate in (project, Compile)).value
-                val sourceMapsEnabled = (finallyEmitSourceMaps in (project, Compile, sjsStage in project)).value
-                files.map(_.data).pair(Path.relativeTo(clientTarget)).map((_, sourceMapsEnabled))
+              Def.taskDyn {
```

Comment:
This can be a `settingDyn` since the dynamic part only depends on the value of a setting:
```suggestion
              Def.settingDyn {
```